File: lib/data/model/ModuleSettings.dart
import 'package:flutter/material.dart';

import 'Module.dart';

class ModuleSettings {
  int id = 0;
  String moduleId;
  bool fuelEmptyNotificationsEnabled;
  bool pumpActivationScheduleEnabled;
  TimeOfDay pumpActivationTime;
  bool pumpShutdownScheduleEnabled;
  TimeOfDay pumpShutdownTime;

  ModuleSettings({
    required this.moduleId,
    required this.fuelEmptyNotificationsEnabled,
    required this.pumpActivationScheduleEnabled,
    required this.pumpShutdownScheduleEnabled,
    this.pumpActivationTime = const TimeOfDay(hour: 9, minute: 0),
    this.pumpShutdownTime = const TimeOfDay(hour: 21, minute: 0)
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'moduleId': moduleId,
      'fuelEmptyNotificationsEnabled': fuelEmptyNotificationsEnabled? 1 : 0,
      'pumpActivationScheduleEnabled': pumpActivationScheduleEnabled? 1 : 0,
      'pumpActivationTime': "${pumpActivationTime.hour}:${pumpActivationTime.minute}",
      'pumpShutdownScheduleEnabled': pumpShutdownScheduleEnabled? 1 : 0,
      'pumpShutdownTime': "${pumpShutdownTime.hour}:${pumpShutdownTime.minute}",
    };
  }
}

File: app/src/main/java/pl/damianhoppe/emodulnotifier/data/model/ModuleSettings.kt
package pl.damianhoppe.emodulnotifier.data.model

import androidx.core.math.MathUtils.clamp
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

const val DEFAULT_FUEL_LEVEL_THRESHOLD_FOR_NOTIFICATION: Int = 20
const val FUEL_LEVEL_MIN: Int = 0
const val FUEL_LEVEL_MAX: Int = 100

@Entity(tableName = "module_settings")
class ModuleSettings (

    @PrimaryKey(autoGenerate = true)
    var id: Long,
    val moduleId: String,
    var fuelEmptyNotificationsEnabled: Boolean = false,
    fuelLevelThresholdForNotification: Int = DEFAULT_FUEL_LEVEL_THRESHOLD_FOR_NOTIFICATION,
    var pumpActivationScheduleEnabled: Boolean = false,
    var pumpActivationTime: Int = 8*60,
    var pumpShutdownScheduleEnabled: Boolean = false,
    var pumpShutdownTime: Int = 20*60,
) {
    @ColumnInfo(defaultValue = "15")
    var fuelLevelThresholdForNotification: Int = fuelLevelThresholdForNotification
        set(value) {
            field = clamp(value, FUEL_LEVEL_MIN, FUEL_LEVEL_MAX)
        }

    companion object {
        fun Default(moduleId: String): ModuleSettings {
            return ModuleSettings(0, moduleId)
        }
    }
}